mozjs/src/conversions.rs

```diff
@@ -166,6 +166,29 @@ pub trait FromJSValConvertible: Sized {
     ) -> Result<ConversionResult<Self>, ()>;
 }
 
+/// A trait to convert `JSVal`s to Rust types inside of Rc wrappers.
+pub trait FromJSValConvertibleRc: Sized {
+    /// Convert `val` to type `Self`.
+    /// If it returns `Err(())`, a JSAPI exception is pending.
+    /// If it returns `Ok(Failure(reason))`, there is no pending JSAPI exception.
+    unsafe fn from_jsval(
+        cx: *mut JSContext,
+        val: HandleValue,
+    ) -> Result<ConversionResult<Rc<Self>>, ()>;
+}
+
+impl<T: FromJSValConvertibleRc> FromJSValConvertible for Rc<T> {
+    type Config = ();
```

Comment:
While I would prefer to use `type Config = T::Config;` here, I was not able to make that compile in Servo's generated bindings. I kept running into compiler errors where passing the `()` config option to FromJSValConvertible for a `Rc<Promise>` where the compiler expected an associated type.

Reply:
Maybe because T (`FromJSValConvertibleRc`) does not have any `type Config` in trait definition.

Reply:
The errors I refer to.iccurred when both traits were defined identically (apart from the Rc return value).